.github/workflows/poll-starter-workflows.yml

```diff
@@ -0,0 +1,80 @@
+name: Poll Starter Workflows
+
```

Comment:
💡 question: so every 6 hours there will be a commit in this repo for new workflows? 

I am keen to possibly understand the use case for it? And then we can follow a different approach for this solution! So whatever repo it is, the Master branch contains the true copy of the AKS workflows?

Reply:
Every 6 hours this workflow executes which checks if there has been an update to the starter workflows housed in their main repository. If there's no new changes, no PR or commit is made. If there's new changes a PR is made suggesting those changes. If the PR is rejected, no PR is made until new changes are made to the starter workflows housed in their main repository.

Reply:
The use case is it informs us if starter repos have changed since we last saw them by making a PR if they have. This lets this repo keep up to date (with security vulnerability fixes and other changes).

Reply:
💡 Thanks for reply, Cool, so will be better to do this, I can add a PR with this, but doing this we can work on whatever the the latest for that repo.

``` *.ts
export async function workOnWhateverLatestProductionCopyis(): Promise<string | undefined> {
     const tempFile = tmp.fileSync({ prefix: "azure-kubernetes-service-", postfix: `.yaml` });

     try {
         const response = await axios.default.get('https://raw.githubusercontent.com/actions/starter-workflows/main/deployments/azure-kubernetes-service.yml');
         fs.writeFileSync(tempFile.name, response.data);
         replaceDeploymentResourceAndClusterName(some_object, tempFile.name);
         return tempFile.name;
     } catch (e) {
         vscode.window.showErrorMessage(`Azure-kubernetes-service Deployment file had following error: ${e}`);
         return undefined;
     }
 }
```

Reply:
💡 So the key question here is will the Yaml file ever contain something which is a breaking change and cause the vscode to fail in this case all I can think off that if `resoruce name` and `cluster name` (placeholder values) are removed then there is nothing to replace. (Thinking out loud I guess then also nothing will break) cc: @peterbom

Reply:
Partner_templates is what they ask us to raise PRs against. 

Pulling the workflows directly seems scary in case there ever is a breaking change. The workflow that I have created essentially does the same thing while giving this repository clear visibility into any changes.

In the case that resource name and cluster name placeholders are replaced it wouldn't technically break anything but it would be helpful to alert us that this change was made (which this workflow does).

Reply:
Ah, okies, so breaking changes is something to watch out for, Thanks this is helping to understand the use-case and associated changes.

**Another Quick Scenario Question** - Lets say there was a change of a line in first 6 hours and then after that just before second polling someone added another small line of change. For with first poll there will be PR-1 with one line of change and then in second round there will be PR-2 with 2 lines of change (including the one in PR-1) so essentially the PR-1 is now redundant - in that scenario we will need to physically go and close the PR-1 and keep PR-2 as real PR?

Reply:
Nice, 👍 sounds good! Thank you 🙏 rest will be leftover yaml changes for placeholder value, all depends if you are comfortable to make those changes or we do it separately.